### PR TITLE
Clean up vroom files to introduce concepts better, add autopep8 coverage

### LINKED ISSUE #11: Clean up codefmt vroom files
The codefmt vroom files are pretty good for test coverage right now but not very friendly and readable as "executable documentation". Compare to the glaive docs at https://github.com/google/vim-glaive/tree/master/vroom as an A+ example. Improvements to consider:
  * Add more introductory text and a "table of contents" in main.vroom. Consider renaming to codefmt.vroom.
  * Split off a separate formatters.vroom file for testing the built-in formatters and consider using very simple dummy formatters in the other files.
  * Cut down on repetition. It's not easy right now to see at a glance what each executable section is testing/demonstrating for all of the verbose syntax.

### PR DESCRIPTION
**Pull request by dbarnett**
Cleans up vroom files so they explain what they're doing more clearly and introduce concepts in a more natural order. Cuts down on a lot of repetition by leaving out details that don't relate to the feature being demonstrated/tested, like using more wildcards in syscall lines and formatting empty buffers in places.

Also adds a lot of coverage and adds testing for autopep8 which previously had none.

Fixes #11.

### CODE REVIEW

**Comment by malcolmr** [comment]
Looks great; CI tests are failing, though?

File: autoload/codefmt.vim [hunk 1]
```diff
@@ -393,3 +393,21 @@ endfunction
...
 endfunction
+
+
+""
+" @private
+" Undoes the effect of @function(#DisableIsAvailableChecksForTesting), allowing
+" FORMATTER.IsAvailable checks to work normally.
+function! codefmt#DontDisableIsAvailableChecksForTesting() abort
+  let s:check_formatters_available = 1
+endfunction
```
Comment (fowles) [inline autoload/codefmt.vim]: Seems like one method that takes an argument would be more natural.  If you want to stick with two methods, you should avoid double negatives.  DontDisable is better written Enable

File: vroom/autopep8.vroom [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  $ 
+
+The name or path of the autopep8 executable can be configured via the
+autopep8_executable flag if the default of "autopep8" doesn't work.
+
+  :Glaive codefmt autopep8_executable='autopep20'
+  :FormatCode autopep8
+  ! autopep20 .*
+  $ 
+  :Glaive codefmt autopep8_executable='autopep8'
```
Comment (fowles) [inline vroom/autopep8.vroom]: Are these glaive commands clearing the cached --version?  Should we use that everywhere?

Comment (dbarnett) [inline vroom/autopep8.vroom]: Good idea. Then I can get rid of `codefmt#ForgetAutopep8VersionForTesting` and just overwrite the flag instead.

File: autoload/codefmt.vim (hunk 1 above)
Comment (dbarnett) [inline autoload/codefmt.vim]: Any suggestions for a name for the single method? `codefmt#OverrideWhetherToDoIsAvailableChecksForTesting({enable})`? Long names should be fine for this as long as they're clear at a glance.

Comment (fowles) [inline autoload/codefmt.vim]: codefmt#SetWhetherToPerformIsAvailableChecksForTesting

**Comment by dbarnett** [comment]
I tracked down the issue with the tests to google/vroom#84. Pretty gnarly issue. Worked around it here by adding placeholder content everywhere.

Also updated the autopep8 version caching and testing hooks as suggested. PTAL.
